### PR TITLE
pythonPackages/yolk: init at 0.4.3

### PR DESCRIPTION
**Pull request by Profpatsch**
pypi inspector

cc @garbas 
btw I made a yas snippet for inserting packages into pythonPackages. It’s quite nice.

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -22212,6 +22212,25 @@ in modules // {
...
+      url = "https://pypi.python.org/packages/source/y/yolk/yolk-${meta.version}.tar.gz";
+      sha256 = "1f6xwx210jnl5nq0m3agh2p1cxmaizawaf3fwq43q4yw050fn1qw";
+    };
+
+    buildInputs = with self; [ nose ];
+
+    meta = {
+      description = "Command-line tool for querying PyPI and Python packages installed on your system.";
+      homepage = "https://github.com/cakebread/yolk";
+      version = "0.4.3";
```
Comment (lancelotsix) [inline pkgs/top-level/python-packages.nix]: The version tag is usually not declared in `meta` but one level above (alongside with `name`). If you have a look at `doc/meta.xml`, you will see this is not part of the expected meta attributes.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -22212,6 +22212,25 @@ in modules // {
...
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/y/yolk/yolk-${meta.version}.tar.gz";
+      sha256 = "1f6xwx210jnl5nq0m3agh2p1cxmaizawaf3fwq43q4yw050fn1qw";
+    };
+
+    buildInputs = with self; [ nose ];
+
+    meta = {
+      description = "Command-line tool for querying PyPI and Python packages installed on your system.";
+      homepage = "https://github.com/cakebread/yolk";
```
Comment (lancelotsix) [inline pkgs/top-level/python-packages.nix]: You do not need the quotes around plain URLs

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -22212,6 +22212,25 @@ in modules // {
...
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/y/yolk/yolk-${meta.version}.tar.gz";
+      sha256 = "1f6xwx210jnl5nq0m3agh2p1cxmaizawaf3fwq43q4yw050fn1qw";
+    };
+
+    buildInputs = with self; [ nose ];
+
+    meta = {
+      description = "Command-line tool for querying PyPI and Python packages installed on your system.";
```
Comment (lancelotsix) [inline pkgs/top-level/python-packages.nix]: descriptions do not have period at the end (see `doc/meta.xml`)

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -22212,6 +22212,25 @@ in modules // {
     };
   };
 
+  yolk = buildPythonPackage rec {
```
Comment (lancelotsix) [inline pkgs/top-level/python-packages.nix]: If this is an application (command line tool), `buildPythonApplication` should be preferred over `buildPythonPackage` and it should be placed independently in the pkgs tree (not in `python-packages.nix` which is used for packages applications rely on).

`git grep buildPythonApplication` should give you many examples.

If you want, there is a pending documentation PR (#13320) that include discussion about this subject.

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (Profpatsch) [inline pkgs/top-level/python-packages.nix]: They removed it! https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

Good to know.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (Profpatsch) [inline pkgs/top-level/python-packages.nix]: Doesn’t hurt, though.

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (Profpatsch) [inline pkgs/top-level/python-packages.nix]: I really should always consult the current nixpkgs documentation, too much has changed from 15.09 …

**Comment by Profpatsch** [comment]
Thanks for the comments!

**Comment by FRidh** [comment]
ping

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (Profpatsch) [inline pkgs/top-level/python-packages.nix]: > If you want, there is a pending documentation PR (#13320) that include discussion about this subject.

That PR was reverted, so there doesn’t seem to be a lot of consenus.

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: I reverted that PR because I merged it before it was ready.

**Comment by Profpatsch** [comment]
@FRidh pong

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -9413,10 +9413,12 @@ in
 
   slowaes = pythonPackages.slowaes;
 
+  twisted = pythonPackages.twisted;
+
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: Why is twisted added here?

Comment (Profpatsch) [inline pkgs/top-level/all-packages.nix]: Because alphabet.

**Pull request merged**